controllers/userController.js

```diff
@@ -7,8 +7,8 @@ const userValidate = require("../services/userVAlidationService");
 exports.signup = async (req, res, next) => {
   try {
     const{userName, email, password} = req.body;
-    const validationErrors = userValidate.validateSignupCredentials(userName, email, password);
-    if (validationErrors.length > 0) {
+    
+    if (!userName || !email || !password) {
       return res.status(422).json({ errors: validationErrors });
```

Comment:
don't call validation error function when the credentials are not coming in req.body, simply return this message "required fields are missing"
run the validationErrors message when fields are coming so then you can validate them according to the rules you define in function